tests/drivers/build_all/ethernet/prj.conf

```diff
@@ -7,3 +7,9 @@ CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_TEST_USERSPACE=y
 
 CONFIG_GPIO=y
+
+CONFIG_SPI_INIT_PRIORITY=50
+CONFIG_ETH_INIT_PRIORITY=60
+CONFIG_ETH_ADIN2111_INIT_PRIORITY=60
+CONFIG_MDIO_INIT_PRIORITY=70
+CONFIG_PHY_INIT_PRIORITY=80
```

Comment:
yikes... how about setting all of them at the same value and let the ordinal do the sequencing?

Reply:
> how about setting all of them at the same value and let the ordinal do the sequencing

Checked such approach, seems to work.

It is definitely possible to get rid of `CONFIG_ETH_ADIN2111_INIT_PRIORITY` in favor of `CONFIG_ETH_INIT_PRIORITY`. Can that be done in this PR or do we need a separate one?

Maybe it is worth to consider modifying `CONFIG_MDIO_INIT_PRIORITY` default to `70`? That should work fine for others since #60816 and #61563.

Reply:
> Checked such approach, seems to work.
> 

Thank you

> It is definitely possible to get rid of `CONFIG_ETH_ADIN2111_INIT_PRIORITY` in favor of `CONFIG_ETH_INIT_PRIORITY`. Can that be done in this PR or do we need a separate one?
> 

I can make the change but I don't have hardware to test it. If you can help, that would be great.

> Maybe it is worth to consider modifying `CONFIG_MDIO_INIT_PRIORITY` default to `70`? That should work fine for others since #60816 and #61563. However, I see that `sam_v71b_xult`, `sam_e70`, `sam4e_xpro` fail.

I was reluctant to change default priorities due to my lack of hardware to test. I need to investigate the sam build failures.